pkg/types/position.go

```diff
@@ -742,11 +751,28 @@ func (p *Position) AddTrade(td Trade) (profit fixedpoint.Value, netProfit fixedp
 
 func (p *Position) UpdateMetrics() {
 	p.Lock()
-	p.updateMetrics()
+	p.updateMetrics(nil)
 	p.Unlock()
 }
 
-func (p *Position) updateMetrics() {
+// SetLastPrice sets the last price of the position
+// this enables the position to calculate the unrealized profit
+func (p *Position) SetLastPrice(price fixedpoint.Value) {
+	p.Lock()
+	defer p.Unlock()
+
+	p.lastPrice = price
+	p.lastPriceTime = time.Now()
+}
+
+func (p *Position) LastPrice() fixedpoint.Value {
+	p.Lock()
+	defer p.Unlock()
+
+	return p.lastPrice
```

Comment:
how do you handle expired last price?

Reply:
If the last update time of the price is over 3 minutes, the price is stale and the unrealized profit metric will not be updated.